packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts

```diff
@@ -25,7 +25,7 @@ export const outOfOfficeCreate = async ({ ctx, input }: TBookingRedirect) => {
 
   const inputStartTime = dayjs(startDate).startOf("day");
   const inputEndTime = dayjs(endDate).endOf("day");
-  const offset = dayjs(inputStartTime).utcOffset();
+  const offset = input.offset;
```

Comment:
Hey @ImBIOS Isn't both gives the same result? I think it would be better to use server tz here .

Reply:
`inputStartTime` doesn't have timezone offset data in it -it's just date and time-, that might be the purpose of the engineer who created the OOO feature sent this offset data to the server, but somehow he doesn't consume it in the server. This offset data is required to make the server know what's the time in UTC+0 -the server timezone-.

The server timezone is UTC+0 if it's in the GitHub Actions by default, and the frontend should be UTC+1 if it follows the playwright config which is `Europe/London`. This makes the `inputStartTime` generated by `dayjs` always 1 day before (1 hour before today's start time) the server time.

CMIIW.

If you want, I am also able to write more edge cases for this OOO feature so we do not just test the functional, but also the edge cases, e. g. different timezones, so we can test this offset data, if you agree I'll work on adding more tests for it by God's will.

Reply:
Hey, we don't want today to be set as the start time; rather, the start date must be in the future. If we use the client-side generated offset, it will allow me to set today as the start time. That's why we use the server tz here. Let's not change it. Let's find a different way to fix the test.

Reply:
@ImBIOS look the above message

Reply:
Don't worry, I've been dedicating myself to working on it, since morning. I found something new and interesting.

Is it expected to be able to set today as startTime in UTC- but it cannot be set today for the startTime if we in UTC+ ???

Reply:
> Don't worry, I've been dedicating myself to working on it, since morning. I found something new and interesting.
> 
> Is it expected to be able to set today as startTime in UTC- (e.g. America/Phoenix) but it cannot be set today for the startTime if we are in UTC+ (e.g. Asia/Jakarta) ??? Is this inconsistent behavior expected?
> 
> I found this interesting behavior when I was trying to add several E2E edge cases on timezone-difference users. And it happens when I use this:
> 
> ```
>   const offset = dayjs(inputStartTime).utcOffset();
> ```
> 
> I'm not testing with `input.offset` yet because you suggested. I revert to using server tz, so I'm working on testing the behavior when using server tz, to make sure it's working as expected.

Something weird, now I can't reproduce this finding. The tests always fail from morning to afternoon, but since 15 minutes ago, these tests always succeeded no matter what timezone I tested it from.

If God wills, I'll keep update to test it. But for now, et's use the server tz, I'll push it now.

Reply:
Sorry for the drama, I've figured out the root cause. On my way to pushing it...

Reply:
All good. You're doing great work

Reply:
wait.. I wannna add some bonus to prevent this kind of bug happen ever again.

Reply:
I added tests against this code for all time zones to make sure no timezone facing this issue.

Reply:
@anikdhabal This bug might be solved and the e2e tests might passed, but for your information... some timezones have a 1-day range (e.g. 3 - 4) as the default value in the modal date range and I noticed some timezones will have a 2-day range (e.g. 3 - 5) as the default value, because it's another issue which -in my opinion- should be a patch bug, I'm not addressing this issue as of right now.